atomix/utils/src/main/java/io/atomix/utils/serializer/Namespace.java

```diff
@@ -260,388 +64,16 @@ public void serialize(final Object obj, final OutputStream stream, final int buf
-    }
-
-    /**
-     * Registers all the class registered to given KryoNamespace.
-     *
-     * @param ns KryoNamespace
-     * @return this
-     */
-    public Builder register(final Namespace ns) {
-
-      if (blocks.containsAll(ns.registeredBlocks)) {
-        // Everything was already registered.
-        LOGGER.debug("Ignoring {}, already registered.", ns);
-        return this;
-      }
-      for (final RegistrationBlock block : ns.registeredBlocks) {
-        this.register(block);
-      }
-      return this;
-    }
-
-    /**
-     * Sets the namespace class loader.
-     *
-     * @param classLoader the namespace class loader
-     * @return the namespace builder
-     */
-    public Builder setClassLoader(final ClassLoader classLoader) {
-      this.classLoader = classLoader;
-      return this;
-    }
-
-    /**
-     * Sets whether backwards/forwards compatible versioned serialization is enabled.
-     *
-     * <p>When compatible serialization is enabled, the {@link CompatibleFieldSerializer} will be
-     * set as the default serializer for types that do not otherwise explicitly specify a
-     * serializer.
-     *
-     * @param compatible whether versioned serialization is enabled
-     * @return this
-     */
-    public Builder setCompatible(final boolean compatible) {
-      this.compatible = compatible;
-      return this;
-    }
-
-    /**
-     * Sets the registrationRequired flag.
-     *
-     * @param registrationRequired Kryo's registrationRequired flag
-     * @return this
-     * @see Kryo#setRegistrationRequired(boolean)
-     */
-    public Builder setRegistrationRequired(final boolean registrationRequired) {
-      this.registrationRequired = registrationRequired;
-      return this;
-    }
-  }
-
-  static final class RegistrationBlock {
-    private final int begin;
-    private final ImmutableList<Pair<Class<?>[], Serializer<?>>> types;
-
-    RegistrationBlock(final int begin, final List<Pair<Class<?>[], Serializer<?>>> types) {
-      this.begin = begin;
-      this.types = ImmutableList.copyOf(types);
-    }
-
-    public int begin() {
-      return begin;
-    }
-
-    public ImmutableList<Pair<Class<?>[], Serializer<?>>> types() {
-      return types;
-    }
-
-    @Override
-    public int hashCode() {
-      return types.hashCode();
-    }
-
-    // Only the registered types are used for equality.
-    @Override
-    public boolean equals(final Object obj) {
-      if (this == obj) {
-        return true;
-      }
-
-      if (obj instanceof RegistrationBlock) {
-        final RegistrationBlock that = (RegistrationBlock) obj;
-        return Objects.equals(this.types, that.types);
-      }
-      return false;
-    }
-
-    @Override
-    public String toString() {
-      return MoreObjects.toStringHelper(getClass())
-          .add("begin", begin)
-          .add("types", types)
-          .toString();
-    }
+  static NamespaceImpl.Builder builder() {
+    return new NamespaceImpl.Builder();
```

Comment:
It would be good not to tie implementation with the interface. So I would move this method out and put it in the `NamespaceImpl`.